File: src/services/microservice.js
class Microservice {
    constructor(name) {
        this.name = name;
        this.protocol = null;
    }

    selectProtocol(protocolType) {
        if (!['REST', 'gRPC', 'GraphQL'].includes(protocolType)) {
            throw new Error("Invalid protocol type. Choose from 'REST', 'gRPC', or 'GraphQL'.");
        }
        this.protocol = protocolType;
        return `${this.name} is using ${this.protocol} protocol.`;
    }

    getDetails() {
        return `Microservice: ${this.name}, Protocol: ${this.protocol ? this.protocol : 'Not selected'}`;
    }
}

// Example usage:
const service1 = new Microservice('User Service');
console.log(service1.selectProtocol('REST'));
console.log(service1.getDetails());

const service2 = new Microservice('Order Service');
console.log(service2.selectProtocol('gRPC'));
console.log(service2.getDetails());

File: src/services/microservice.py
class Microservice:
    """
    A class representing a microservice with methods for selecting a communication protocol.
    """
    def __init__(self, name):
        self.name = name
        self.protocol = None

    def select_protocol(self, protocol_type):
        """Selects a communication protocol for the microservice."""
        if protocol_type not in ['REST', 'gRPC', 'GraphQL']:
            raise ValueError("Invalid protocol type. Choose from 'REST', 'gRPC', or 'GraphQL'.")
        self.protocol = protocol_type
        return f"{self.name} is using {self.protocol} protocol."

    def get_details(self):
        """Returns the details of the microservice including its protocol."""
        return f"Microservice: {self.name}, Protocol: {self.protocol if self.protocol else 'Not selected'}"

# Example usage:
if __name__ == '__main__':
    service1 = Microservice('User Service')
    print(service1.select_protocol('REST'))
    print(service1.get_details())
    
    service2 = Microservice('Order Service')
    print(service2.select_protocol('gRPC'))
    print(service2.get_details())